### PR TITLE
Run workflow form

### PR DESCRIPTION
**Pull request by guerler**
This is the first PR for implementing and prepping the form classes to accommodate the new run workflow form. The new form is disabled by default (run_workflow_toolform_upgrade).

### CODE REVIEW

File: lib/galaxy/tools/parameters/basic.py [hunk 1]
```diff
@@ -347,7 +341,10 @@ def from_html( self, value, trans=None, other_values={} ):
         try:
             return int( value )
         except:
-            if not value and self.optional:
+            if isinstance( value, basestring ):
```
Comment (jmchilton) [inline lib/galaxy/tools/parameters/basic.py:347]: Can you catch a more specific exception here an add comment about the $.

File: lib/galaxy/webapps/galaxy/api/workflows.py [hunk 2]
```diff
@@ -136,12 +136,12 @@ def create(self, trans, payload, **kwd):
             'from_history_id',
             'shared_workflow_id',
             'workflow',
-        ] ).intersection( payload )
-        if len( ways_to_create ) == 0:
+        ] )
+        if len( ways_to_create.intersection( payload ) ) == 0:
```
Comment (jmchilton) [inline lib/galaxy/webapps/galaxy/api/workflows.py:139]: Are you running workflows via this endpoint and not the POSTing to invocations? Running workflows on this endpoint has been deprecated.

Comment (guerler) [inline lib/galaxy/webapps/galaxy/api/workflows.py:139]: This is a fix for the deprecated endpoint. The form itself is sending the workflow request to the invocations endpoint: https://github.com/guerler/galaxy/blob/7ffbd0907658ee071fe32393c03bdfee4b6b740c/client/galaxy/scripts/mvc/tool/tool-form-composite.js#L268

Comment (jmchilton) [inline lib/galaxy/webapps/galaxy/api/workflows.py:139]: Awesome, thanks!

**Comment by jmchilton** [comment]
There are three failing API tests that seem like legimate regressions. https://jenkins.galaxyproject.org/job/docker-api/1494/#showFailuresLink

**Comment by guerler** [comment]
Thanks for looking into it.

**Comment by guerler** [comment]
@galaxybot test this

**Comment by bgruening** [comment]
@guerler we need a rebase here and have 4 failing tests.

**Comment by martenson** [comment]
@galaxybot test this
The 4 failing TS tests are unrelated to this PR, they should be fixed with this run.
edit: not fixed by rerun but definitely unrelated

**Comment by bgruening** [comment]
- it would be nice if the entire tool-header would be click able and open/close a tool
- on a large screen the tools are on the right side of the main frame ... center or justify would be nice
- on the other hand, now that we have more space left and right to the tool boxes, some arrows left and right indicating from which input belongs to which output would be great

Usability feels very good! Great work Aysam!

Anyone wants to test it? Use this prepared Docker Image:

```
docker run -p 8080:80 -i -t bgruening/galaxy-stable:guerler
```

**Comment by martenson** [comment]
@guerler This is the fix for the conflict. I tried to create a PR against your branch but because it is way behind dev (and in conflict :D) it caused trouble so I am sending it at least like this.

https://github.com/guerler/galaxy/commit/3eef069ea0795e8d06386b1acbae9eb1d4502617#diff-fae872b63d46cbf39f4d2749c699f8e0R12

The whole thing with `dev` merged is in https://github.com/martenson/galaxy/commits/fix-js-imports (the `fix-js-imports` branch)

**Comment by martenson** [comment]
Also I am getting 

```
File '/Users/marten/devel/git/galaxy/database/compiled_templates/workflow/run.mako.py', line 91 in render_body
  step_model[ 'step_id' ] = step.id
TypeError: 'NoneType' object does not support item assignment
```

when trying to access the WF run form (no matter if the dev-merged version or @guerler's)
What am I doing wrong?

**Comment by jmchilton** [comment]
@guerler and I discussed this and have decided to mirror the effort of the new tool form (keeping it optional by default initially). We will merge this today (for the exposure and because "97 changed files  with 1,374 additions and 1,219 deletions." stales quickly) and then create an issue with things that need to be done before this can be the primary workflow form.

**Comment by martenson** [comment]
@jmchilton :+1:

**Pull request merged**